onnxruntime/core/providers/coreml/coreml_execution_provider.cc

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "coreml_execution_provider.h"
+
+#include "core/framework/allocatormgr.h"
+#include "core/framework/compute_capability.h"
+#include "core/graph/graph_viewer.h"
+#include "core/session/onnxruntime_cxx_api.h"
+
+#include "model/model.h"
+#include "model/host_utils.h"
+#include "builders/helper.h"
+#include "builders/model_builder.h"
+
+namespace onnxruntime {
+
+constexpr const char* COREML = "CoreML";
+
+CoreMLExecutionProvider::CoreMLExecutionProvider()
+    : IExecutionProvider{onnxruntime::kCoreMLExecutionProvider} {
+  AllocatorCreationInfo device_info(
+      [](int) {
+        return onnxruntime::make_unique<CPUAllocator>(OrtMemoryInfo(COREML, OrtAllocatorType::OrtDeviceAllocator));
+      });
+
+  InsertAllocator(CreateAllocator(device_info));
+
+  AllocatorCreationInfo cpu_memory_info(
+      [](int) {
+        return onnxruntime::make_unique<CPUAllocator>(
+            OrtMemoryInfo(COREML, OrtAllocatorType::OrtDeviceAllocator, OrtDevice(), 0, OrtMemTypeCPUOutput));
+      });
+
+  InsertAllocator(CreateAllocator(cpu_memory_info));
+}
+
+CoreMLExecutionProvider::~CoreMLExecutionProvider() {}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+CoreMLExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view,
+                                       const std::vector<const KernelRegistry*>& /*kernel_registries*/) const {
+  std::vector<std::unique_ptr<ComputeCapability>> result;
```

Comment:
Would be good to make sure that GetCapability/Compile has the same setup as the NNAPI EP so that it can be built and executed on any platform for ORT model generation from early on.

Reply:
It's a pending work item 

---
In reply to: [563461434](https://github.com/microsoft/onnxruntime/pull/6392#discussion_r563461434) [](ancestors = 563461434)